flang/lib/Lower/OpenMP/Utils.cpp

```diff
@@ -537,6 +681,12 @@ void insertChildMapInfoIntoParent(
           mapOperands[std::distance(mapSyms.begin(), parentIter)]
               .getDefiningOp());
 
+      // Once explicit members are attached to a parent map, do not also invoke
+      // a declare mapper on it, otherwise the mapper would remap the same
+      // components leading to duplicate mappings at runtime.
+      if (!indices.second.memberMap.empty() && mapOp.getMapperIdAttr())
```

Comment:
Perhaps a dumb question, would this prevent mapping of `scalar` in the below example (excuse the pseudo code!):

   type :: dtype2
    integer, allocatable :: scalar
   end type :: dtype2

    type :: dtype
    real(4) :: i 
    dtype2 :: dtype2_v
    end type dtype

    dtype :: dtype_v 

!$omp target map(dtype_v%i, dtype_2_v)
!$omp end target

As I think we'd still want to map it in these scenarios as far as I'm aware at least, Michael Klemm may know better.

Reply:
Ah, makes sense, thank you for explaining!